tests/operators/test_virtualenv.py

```diff
@@ -339,3 +356,23 @@ def test__release_venv_lock_current_process(tmpdir):
     )
     assert venv_operator._release_venv_lock() is None
     assert not lockfile.exists()
+
+
+@pytest.mark.skipif(
+    AIRFLOW_VERSION < Version("2.5"),
+    reason="This error is only reproducible with dag.test, which was introduced in Airflow 2.5",
+)
+@pytest.mark.integration
+def test_integration_virtualenv_operator(caplog):
+    """
+    Confirm we're using the correct dbt command to run with virtualenv.
+    """
+    from airflow.models.dagbag import DagBag
+
+    dag_bag = DagBag(dag_folder=DAGS_FOLDER, include_examples=False)
+    dag = dag_bag.get_dag("example_virtualenv_mini")
+
+    dag.test()
+
+    assert "Trying to run the command:\n ['/tmp/persistent-venv2/bin/dbt', 'deps'" in caplog.text
```

Comment:
nice testing strategy, thanks for showing this

Reply:
@pankajkoti, my initial attempts involved mocking `run_command`, `run_subprocess`, and even `FullOutputSubprocessResult`. Unfortunately, as described in the original PR (https://github.com/astronomer/astronomer-cosmos/pull/1252#issue-2582912160), all these things behaved very well when we tried to run the DAG using `from tests.utils import test_dag as run_test_dag`, because the issue is a bit deeper in how Airflow pickles/runs things.

Since I could reproduce by running `dags test`, I decided to give the built-in Airflow `dag.test` a try, and I was glad it worked. It was a bit annoying to find a way to test this, but I'm happy we won't have any regressions.